### PR TITLE
[feat] support smc proxy, add dtb in vcpu_create_cfg

### PR DESCRIPTION
**Pull request by hky1999**

### CODE REVIEW

File: src/exception.rs [hunk 1]
```diff
@@ -71,7 +71,12 @@ core::arch::global_asm!(
 ///
 pub fn handle_exception_sync(ctx: &mut TrapFrame) -> AxResult<AxVCpuExitReason> {
     match exception_class() {
-        Some(ESR_EL2::EC::Value::DataAbortLowerEL) => handle_data_abort(ctx),
+        Some(ESR_EL2::EC::Value::DataAbortLowerEL) => {
+            let elr = ctx.exception_pc();
```
Comment (luodeb) [inline src/exception.rs:75]: Why not put this in `handle_data_abort`?

File: src/exception.rs [hunk 2]
```diff
@@ -96,6 +101,22 @@ pub fn handle_exception_sync(ctx: &mut TrapFrame) -> AxResult<AxVCpuExitReason>
             })
         }
         Some(ESR_EL2::EC::Value::TrappedMsrMrs) => handle_system_register(ctx),
+        Some(ESR_EL2::EC::Value::SMC64) => {
```
Comment (luodeb) [inline src/exception.rs:104]: Better to write it as a function as well, such as `handle_smc64_exception`

File: src/exception.rs (hunk 1 above)
Comment (hky1999) [reply src/exception.rs:75]: I don't think it is necessary. 

For some exceptions,(e.g. DataAbortLowerEL, SMC64) we need to manually modify the elr. 

it is more intuitive to put these logics together.

Comment (luodeb) [reply src/exception.rs:75]: OK

**Review by luodeb — APPROVED** [review]

**Pull request merged**